internal/controller/management_controller.go

```diff
@@ -100,6 +101,59 @@ func (r *ManagementReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 	return r.update(ctx, management)
 }
 
+func (r *ManagementReconciler) getRequestedProvidersList(ctx context.Context, management *kcm.Management) ([]kcm.Provider, error) {
+	list := slices.Clone(management.Spec.Providers)
+
+	existingProviders := make(map[string]struct{}, len(list))
+	for _, provider := range list {
+		existingProviders[provider.Name] = struct{}{}
+	}
+
+	var objList kcm.PluggableProviderList
+
+	if err := r.Client.List(ctx, &objList, &client.ListOptions{
+		LabelSelector: labels.SelectorFromSet(labels.Set{
+			kcm.GenericComponentNameLabel: kcm.GenericComponentLabelValueKCM,
+		}),
+	}); err != nil {
+		return nil, err
+	}
+
+	for _, el := range objList.Items {
+		if _, exists := existingProviders[el.Name]; !exists {
+			list = append(list,
+				kcm.Provider{
+					Name:      el.Name,
+					Component: el.Spec.Component,
+				},
+			)
+
+			existingProviders[el.Name] = struct{}{}
```

Comment:
no need to add: the objects list has unique names (due to cluster scope), hence if a value is not presented, it won't be duplicated. unless there are collisions between in-tree and out-of-tree components having the same name. in that case, we should report an error to the condition and stop reconciling

Reply:
This ability is done via the mgmt, I mean, if there is by any change for some reason one `pprov` object named `cluster-api-provider-aws` (so, out-of-tree provider) which we have in-tree, and the values from the OOT are required, then one can just create a new `ProviderTemplate` and then reference it as a custom in the `mgmt.spec`. So to my understanding, there is no need in such ability